src/nupic/torch/modules/sparse_weights.py

```diff
@@ -134,6 +138,18 @@ class SparseWeights(SparseWeightsBase):
     def __init__(self, module, weight_sparsity=None, sparsity=None,
                  allow_extremes=False):
         assert len(module.weight.shape) == 2, "Should resemble a nn.Linear"
+
+        # Add checks to ensure there is no incompatibility
+        if sparsity is None:
+            sparsity = 1 - weight_sparsity
+        if weight_sparsity is None:
+            weight_sparsity = 1 - sparsity
+        if sparsity is not None and weight_sparsity is not None:
```

Comment:
This condition is always `true`

Reply:
Why would it always be True? They are both set to None as default.

Reply:
If they are `None` you are changing their values in lines 144 and 146